models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -45,11 +45,13 @@ abstract class AbstractRelations extends Data implements
     public $pathFormatterClass;
 
     /**
-     * @return array
+     * @return array[
+     *  'classes' => string,
+     * ]
      */
     public function getClasses()
     {
-        return $this->classes;
+        return $this->classes ?: [];
```

Comment:
I think this could be done also for `getDocumentTypes` and `getAssetTypes`

We can also add a default value `[]` to the properties

Reply:
But `getDocumentTypes()` and `getAssetTypes()` does not exist in `AbstractRelations`. So where do you think we could add this default value?

If we can set the default value to the property I cannot judge. I am always cautious doing that because Pimcore decides on a lot of places whether to load or not to load certain data based on a property being an (empty) array or null. Are you really sure it is ok to set the daefult value of `$classes` property in AbstractRelations to `[]`?

Reply:
This methods exists in ManyToManyRelation and ManyToOneRelation.

If you create the fields in the admin interface this properties will filled with arrays. But if you create it via script it can be that this properties are null. So I would prefer to set it with a default value. I think that can't hurt.

In 7.0 we can protect this properties and we can make the correct type in the setter.
Right @weisswurstkanone :) ?

Reply:
Yes also the default value for the `$classes` property in `AbstractRelations`

Reply:
Also a default value for the `objectsAllowed` property in `ManyToManyRelation`. Than I think its perfect :)

Reply:
Thanks for your assistance and feedback @blankse!